module/module.py

```diff
@@ -176,15 +178,76 @@ def get_state_points(data, name, tags={}):
 
         return points
 
+    # Prepare service cache
+    def manage_initial_service_status_brok(self, b):
+        host_name = b.data['host_name']
+        service_description = b.data['service_description']
+        service_id = host_name + "/" + service_description
+        logger.info("[] got initial service status: %s", service_id)
+
+        if host_name not in self.hosts_cache:
+            logger.error(
+                """[influxdb broker] initial service status,
+                host is unknown: %s.""",
+                service_id)
+            return
+
+        self.services_cache[service_id] = {}
+        if '_INFLUX_TAG' in b.data['customs']:
+            tag_dict = {}
+            tags = b.data['customs']['_INFLUX_TAG'].split(',')
+            for tag in tags:
+                key, value = tag.split(":")
+                tag_dict[key] = value
+            self.services_cache[service_id]['_INFLUX_TAG'] = tag_dict
+
+        logger.debug("[influxdb broker] initial service status received: %s",
+                     service_id)
+
+    # Prepare host cache
+    def manage_initial_host_status_brok(self, b):
+        host_name = b.data['host_name']
+        logger.info("[influxdb broker] got initial host status: %s", host_name)
+
+        self.hosts_cache[host_name] = {}
+        if '_INFLUX_TAG' in b.data['customs']:
+            tag_dict = {}
+            tags = b.data['customs']['_INFLUX_TAG'].split(',')
+            for tag in tags:
+                key, value = tag.split(":")
+                tag_dict[key] = value
+            self.hosts_cache[host_name]['_INFLUX_TAG'] = tag_dict
+
+        logger.debug("[influxdb broker] initial host status received: %s",
+                     host_name)
+
     # A service check result brok has just arrived,
     # we UPDATE data info with this
     def manage_service_check_result_brok(self, b):
         data = b.data
 
+        host_name = data['host_name']
+        service_description = data['service_description']
+        service_id = host_name + "/" + service_description
+        logger.debug("[influxdb broker] service check result: %s", service_id)
+
+        if service_id not in self.services_cache:
+            logger.warning(
+                """[influxdb broker] received service check result
+                for an unknown service: %s""",
+                service_id)
+            return
```

Comment:
- [ ] Why would we return? We can still process the brok, it just won't have the tags.